src/yacs-web/src/pages/MapDates.vue

```diff
@@ -0,0 +1,142 @@
+                        </template>
+                        <template v-slot:cell(semesterPartName)="data">
+                            <input type="text" class="form-control" :placeholder="data.item.display_string" name="semester_part_name" />
+                        </template>
+                    </b-table>
+                    <b-pagination
+                        v-model="currentPage"
+                        :total-rows="subsemesters.filter(x => x.semester_name === standardSemesterName).length"
+                        :per-page="perPage"
+                        :aria-controls="`edit${standardSemesterName.replace(' ', '')}Tbl`"
+                        align="center"
+                    />
+                </b-container>
+                <button type="submit" :class="{
+                    'btn': true,
+                    'btn-primary': true,
+                    'my-2': true,
+                    'w-50': true,
+                    'success': success,
+                    'fail': fail
+                }"
+                @animationend="resetIndicators"
+                >
+                    Submit <b-spinner v-show="loading" />
+                </button>
+            </form>
+        </section>
+    </b-container>
+</template>
+
+<script>
+import { mapDateRangeToSemesterPart as postDateMapping } from '@/services/AdminService';
+import { getSubSemesters } from '@/services/YacsService';
+import { readableDate, standardDate } from '../utils';
+
+export default {
+    name: "MapDates",
+    props: {},
+    components: {},
+    data () {
+        return {
+            perPage: 3,
+            currentPage: 1,
+            subsemesters: [],
+            displayedColumns: ['dateRange', 'semesterPartName'],
+            standardSemesterNames: [],
+            loading: false,
+            success: false,
+            fail: false
+        }
+    },
+    methods: {
+        onSubmit(event) {
+            if (!this.loading) {
+                this.loading = true;
+                let formData = new FormData(event.target);
+                postDateMapping(formData).then(res => {
+                    console.log(res);
+                    this.success = true;
+                })
+                .catch(error => {
+                    console.log(error);
+                    this.fail = true;
+                })
+                .finally(() => {
+                    this.loading = false;
+                });
+            }
+        },
+        formatDateRange(date1, date2) {
+            return `${readableDate(date1)} - ${readableDate(date2)}`;
+        },
+        standardDate,
+        resetIndicators() {
+            this.fail = false;
+            this.success = false;
+        }
+    },
+    created() {
+        getSubSemesters().then(subsemesters => {
+            this.subsemesters = subsemesters;
+            this.standardSemesterNames = new Set(subsemesters.map(subsemester => subsemester.semester_name));
```

Comment:
Particular reason for using a `Set` for `standardSemesterNames`?

Reply:
I added the property `subsemester.semester_name` which will either be `SUMMER|FALL|SPRING {YEAR}`. Date range to name inputs are grouped under these possible semester names. So, I make sure there's no duplicate entries for these possible semester names, otherwise in the UI you'd see ("SUMMER 2020" [all date ranges pertaining to this semester]) x (repeat count)

Reply:
OH its semester_name I'm dumb lol. Good solution